src/main/java/com/arpnetworking/clusteraggregator/ClusterStatusCache.java

```diff
@@ -109,13 +128,51 @@ public Receive createReceive() {
                         }
                     }
                 })
+                .match(ShardRegion.ClusterShardingStats.class, shardingStats -> {
+                    LOGGER.debug()
+                            .setMessage("Received shard statistics")
+                            .addData("regionCount", shardingStats.getRegions().size())
+                            .log();
+                    final Map<String, Long> shardsPerAddress = Maps.newHashMap();
+                    final Map<String, Long> actorsPerAddress = Maps.newHashMap();
+                    for (final Map.Entry<Address, ShardRegion.ShardRegionStats> entry : shardingStats.getRegions().entrySet()) {
+                        final String address = entry.getKey().hostPort();
+                        for (final Map.Entry<String, Object> stats : entry.getValue().getStats().entrySet()) {
+                            final long currentShardCount = shardsPerAddress.getOrDefault(address, 0L);
```

Comment:
Since this is independent of the actual values you can hoist it outside of the loop and avoid repeatedly incrementing:
```
shardsPerAddress.put(address, entry.getValue().getStats().size());
```

Reply:
That's quite the brain fart. Fixed.